build.gradle

```diff
@@ -1,114 +1,147 @@
 def railsIcon = 'rails.ico'
 def railsCopyright = 'GPL'
 
 // Key variables
-def keyAlias = 'rails18xx'
+//def keyAlias = 'rails18xx'
 // def keyAlias = 'stefanfrey'
 
 // Build directories
-def webstartBuildDir = "${buildDir}/webstart/build"
-def webstartDestDir = "${buildDir}/webstart/signed"
-def keystoreFile = '../../keystore/rails.keys'
+//def webstartBuildDir = "${buildDir}/webstart/build"
+//def webstartDestDir = "${buildDir}/webstart/signed"
+//def keystoreFile = '../../keystore/rails.keys'
 def buildFilesDir = 'src/main/resources/build'
-def resourceFilesDir = 'src/main/resources'
-def versionFile = "version.number"
+//def resourceFilesDir = 'src/main/resources'
+//def versionFile = "version.number"
 
 // Sourceforge variables
-def sourceforgeJnlpPath = "http://${railsCodebase}/webstart"
-def sourceforgeUploadPath = 'web.sourceforge.net/home/project-web/rails/htdocs/webstart'
-def sourceforgeJnlpFileName = 'rails.jnlp'
-def sourceforgeUserName = 'stefanfrey'
+//def sourceforgeJnlpPath = "http://${railsCodebase}/webstart"
+//def sourceforgeUploadPath = 'web.sourceforge.net/home/project-web/rails/htdocs/webstart'
+//def sourceforgeJnlpFileName = 'rails.jnlp'
+//def sourceforgeUserName = 'stefanfrey'
 
-// use version number from property file
-propFileLoader.load(file("${buildFilesDir}/${versionFile}"))
+dependencies {
+    implementation 'commons-io:commons-io:2.6'
+    implementation 'com.github.vlsi.mxgraph:jgraphx:4.1.0'
+    implementation 'org.jgrapht:jgrapht-core:1.4.0'
+    implementation 'org.jgrapht:jgrapht-ext:1.4.0'
+    implementation 'org.slf4j:slf4j-api:1.7.30'
+    implementation 'org.apache.xmlgraphics:batik-bridge:1.12'
+    implementation 'org.apache.xmlgraphics:batik-transcoder:1.12'
+    implementation 'org.apache.xmlgraphics:batik-svggen:1.12'
+    implementation 'org.apache.xmlgraphics:batik-ext:1.12'
+    implementation 'org.apache.xmlgraphics:batik-swing:1.12'
+    implementation 'javazoom:jlayer:1.0.1'
+    implementation 'ch.qos.logback:logback-classic:1.2.3'
+    implementation 'com.google.guava:guava:23.0'
+    implementation 'net.java.dev.designgridlayout:designgridlayout:1.11'
+    implementation 'org.dockingframes:docking-frames-common:1.1.1'
+    implementation 'com.googlecode.combinatoricslib:combinatoricslib:2.2'
+    //implementation 'org.openjfx:javafx-controls:14'
+    implementation 'org.apache.httpcomponents:httpclient:4.5.12'
+    implementation 'org.apache.commons:commons-lang3:3.9'
+
+    testImplementation 'junit:junit:4.13'
+    testImplementation 'org.easytesting:fest-assert-core:2.0M10'
+    testImplementation 'org.easytesting:fest-util:1.2.5'
+    testImplementation 'org.mockito:mockito-core:3.3.3'
+}
 
+javafx {
+    version = "14"
+    modules = [ 'javafx.controls' ]
+}
 
-dependencies {
-  compile fileTree(dir: 'libs/main', include: "**/*.jar")
-  testCompile fileTree(dir: 'libs/test', include: "**/*.jar")
+sourceCompatibility = 1.8
+targetCompatibility = 1.8
```

Comment:
Maybe better `JavaVersion.VERSION_1_8`

Reply:
Sure, but convention-wise you'll find that most gradle build script declare it as it is-- easier to read and switching to the constant doesn't really buy us anything. But again, doesn't really matter to me!

Reply:
My reason for the suggestion is because it is safer. When specifying a number you as a developer need to know what is valid otherwise you can get strange runtime errors. When you use an enum or a static field instead you have compile-time checking that detects wrong java versions (e.g. `1.11` vs `11`)